core/mutation/mutation_test.go

```diff
@@ -0,0 +1,264 @@
+package mutation
+
+import (
+	"database/sql"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/google/keytransparency/core/transaction"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	tpb "github.com/google/keytransparency/core/proto/keytransparency_v1_types"
+	"github.com/google/trillian"
+)
+
+const (
+	logID = 0
+	mapID = 0
+)
+
+func signedKV(t *testing.T, start, end int) []*tpb.SignedKV {
+	if start > end {
+		t.Fatalf("start=%v > end=%v", start, end)
+	}
+	kvs := make([]*tpb.SignedKV, 0, end-start)
+	for i := start; i <= end; i++ {
+		kvs = append(kvs, &tpb.SignedKV{
+			KeyValue: &tpb.KeyValue{
+				Key:   []byte(fmt.Sprintf("key_%v", i)),
+				Value: []byte(fmt.Sprintf("value_%v", i)),
+			}})
+	}
+	return kvs
+}
+
+func prepare(t *testing.T, mutations *fakeMutation, fakeMap *fakeTrillianMapClient) {
+	createEpoch(t, mutations, fakeMap, 1, 1, 6)
+	createEpoch(t, mutations, fakeMap, 2, 7, 10)
+}
+
+func createEpoch(t *testing.T, mutations *fakeMutation, fakeMap *fakeTrillianMapClient, epoch int64, start, end int) {
+	kvs := signedKV(t, start, end)
+	for _, kv := range kvs {
+		if _, err := mutations.Write(nil, kv); err != nil {
+			t.Fatalf("mutations.Write failed: %v", err)
+		}
+	}
+	fakeMap.tmap[epoch] = &trillian.SignedMapRoot{
+		Metadata: &trillian.MapperMetadata{
+			HighestFullyCompletedSeq: int64(end),
+		},
+		MapRevision: epoch,
+	}
+}
+
+func TestGetMutations(t *testing.T) {
+	ctx := context.Background()
+	fakeMutations := &fakeMutation{}
+	fakeMap := newFakeTrillianMapClient()
+	prepare(t, fakeMutations, fakeMap)
+
+	for _, tc := range []struct {
+		description string
+		epoch       int64
+		token       string
+		pageSize    int32
+		mutations   []*tpb.SignedKV
+		nextToken   string
+		success     bool
+	}{
+		{"working case complete epoch 1", 1, "", 6, signedKV(t, 1, 6), "", true},
+		{"working case complete epoch 2", 2, "", 4, signedKV(t, 7, 10), "", true},
+		{"working case partial epoch 1", 1, "", 4, signedKV(t, 1, 4), "4", true},
+		{"working case partial epoch 2", 2, "", 3, signedKV(t, 7, 9), "9", true},
+		{"working case larger page size", 1, "", 10, signedKV(t, 1, 6), "", true},
+		{"working case with page token", 1, "4", 2, signedKV(t, 5, 6), "", true},
```

Comment:
Please add a case for when a token is provided as well. It looks like the existing tests exercise this only for the first page.